addons/controllers/testdata/test-cpi-paravirtual.yaml

```diff
@@ -0,0 +1,26 @@
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: Cluster
+metadata:
+  name: test-cluster-cpi-paravirtual
+  namespace: default
+spec:
+  infrastructureRef:
+    kind: VSphereCluster
+  clusterNetwork:
+    pods:
+      cidrBlocks: [ "192.168.0.0/16","fd00:100:96::/48" ]
+---
+apiVersion: cpi.tanzu.vmware.com/v1alpha1
+kind: CPIConfig
+metadata:
+  name: test-cluster-cpi-paravirtual
+  namespace: default
+spec:
+  vsphereCPI:
+    mode: vsphereParavirtualCPI
+    clusterAPIVersion: "cluster.x-k8s.io/v1beta1"
```

Comment:
Have you found the way to consolidate these paravirtual related field as derived fields?

Reply:
let's consolidate it in a separate PR, before we making this one even larger =)